src/ape_pm/compiler.py

```diff
@@ -21,19 +21,16 @@ def compile(self, filepaths: List[Path]) -> List[ContractType]:
         for path in filepaths:
             with path.open() as f:
                 data = json.load(f)
-            if "manifest" in data:
-                manifest = PackageManifest.from_dict(data)
 
-                if manifest.contractTypes:  # type: ignore
-                    for c in manifest.contractTypes.values():
-                        c.sourcePath = path
-                        contract_types.append(c)
+            if not isinstance(data, list):
+                raise  # Not a valid ABI interface JSON file
 
             else:
                 contract_types.append(
                     ContractType(  # type: ignore
                         contractName=path.stem,
                         abi=data,
+                        sourceId=str(path),
                         sourcePath=path,
```

Comment:
You removed the `sourcePath` attribute from the dataclass, so I think this can be removed.

Reply:
```suggestion
```